nicos_ess/devices/epics/pva/monitor_motor.py

```diff
@@ -252,12 +284,64 @@ def _get_status_message(self):
         """
         Get the status message from the motor if the PV exists.
 
-        :return: The status message if it exists, otherwise an empty string.
+        :return: returns message to display, otherwise an
+        empty string.
+        """
+        epics_msg_pvs = self._parse_epics_message_pvs()
+        if epics_msg_pvs[self.SEVR]:
+            return self._get_epics_err_info(epics_msg_pvs)
+        else:
+            return epics_msg_pvs[self.MSG_TXT]
+
+    def _check_ioc_controller_communication(self):
         """
-        if not self.errormsgpv:
+        Check if there is communication between the IOC and controller.
+
+        :return: returns communication error message if there is no connection,
+        otherwise returns an empty string.
+        """
+        epics_msg_pvs = self._parse_epics_message_pvs()
+        if epics_msg_pvs[self.SEVR] == self.INVALID_SEVR and \
+                epics_msg_pvs[self.STAT] == self.COMM_STAT:
+            return self._get_epics_err_info(epics_msg_pvs)
+        else:
             return ''
 
-        return self._get_pv('errormsgpv', as_string=True)
+    def _parse_epics_message_pvs(self):
+        """
+        Parses the epics status message PVs and gives a dictionary containing
+        these in string format.
+
+        :return: returns a dictionary containing all the information regarding
+        status messages from EPICS.
+        """
+        epics_msg_pvs = {self.SEVR: "",
+                         self.STAT: "",
+                         self.MSG_TXT: ""}
+        if self.errormsgpv:
+            epics_msg_pvs[self.MSG_TXT] = \
+                self._get_pv('errormsgpv', as_string=True)
+        if self.error_severity_pv:
+            epics_msg_pvs[self.SEVR] = \
+                self._get_pv('error_severity_pv', as_string=True)
+        if self.error_status_pv:
+            epics_msg_pvs[self.STAT] = \
+                self._get_pv('error_status_pv', as_string=True)
+        return epics_msg_pvs
+
+    def _get_epics_err_info(self, error):
+        """
+        Sends EPICS error information to the NICOS logger.
+
+        :return: returns an error message string.
+        """
+        session.log.error([f'EPICS ERROR MESSAGE: '
```

Comment:
```suggestion
        self.log.error([f'EPICS ERROR MESSAGE: '
```
Then you don't need to import session